lib/rubocop/cop/naming/variable_number.rb

```diff
@@ -102,6 +102,19 @@ class VariableNumber < Base
 
         MSG = 'Use %<style>s for %<identifier_type>s numbers.'
 
+        def max_allowed_letters_before_number
+          cop_config.fetch('MaxAllowedLettersBeforeNumber', 0)
+        end
+
+        def valid_name?(node, name, given_style = style)
+          return true if super
+
+          return false unless given_style == :snake_case
+          return false unless max_allowed_letters_before_number.to_i.positive?
+
+          name.match?(/(?:_|\b)\D{0,#{max_allowed_letters_before_number}}\d+\z/)
```

Comment:
For readability, this could also be:
```suggestion
          super || (
            given_style == :snake_case && max_allowed_letters_before_number.to_i.positive? &&
            name.match?(/(?:_|\b)\D{0,#{max_allowed_letters_before_number}}\d+\z/)
          )
```